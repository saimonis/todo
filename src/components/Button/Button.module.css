.btn {
  align-self: stretch;
  padding: 0 5px;
  border: 1px solid grey;
  margin-left: 10px;
  cursor: pointer;
}
