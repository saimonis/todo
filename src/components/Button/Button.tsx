import React, { SyntheticEvent } from "react";
import styles from "./Button.module.css";

export default function ({
  text,
  onClick,
}: {
  text: string;
  onClick: (e: SyntheticEvent) => void;
}) {
  return (
    <button className={styles.btn} onClick={onClick}>
      {text}
    </button>
  );
}
