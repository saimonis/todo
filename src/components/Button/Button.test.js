import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("Buttons test", () => {
  const mockFn = jest.fn();
  const btn = shallow(<Button text="btnText" onClick={mockFn} />);
  it("The button has right text", () => {
    expect(btn.text()).toEqual("btnText");
  });

  it("Function from props calls", () => {
    btn.simulate("click");
    expect(mockFn).toBeCalledTimes(1);
  });
  it("renders properly", () => {
    expect(btn.debug()).toMatchSnapshot();
  });
});
