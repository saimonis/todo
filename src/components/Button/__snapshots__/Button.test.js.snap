// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons test renders properly 1`] = `
"<button className=\\"btn\\" onClick={[Function: mockConstructor]}>
  btnText
</button>"
`;
